src/components/tooltip/Tooltip.tsx

```diff
@@ -126,10 +126,26 @@ class Tooltip extends React.Component<TooltipProps, State> {
         this.state = { isShown: !!props.isShown, hasRendered: false, wasClosedByUser: false };
     }
 
+    isControlled = () => {
+        const { isShown: isShownProp } = this.props;
+        return typeof isShownProp !== 'undefined';
+    };
+
     componentDidMount() {
         this.setState({ hasRendered: true });
     }
 
+    componentDidUpdate(prevProps: TooltipProps, prevState: State) {
+        // Reset wasClosedByUser state when isShown transitions from false to true
+        if (this.isControlled()) {
+            if (!prevProps.isShown && this.props.isShown) {
+                this.setState({ wasClosedByUser: false });
+            }
+        } else if (!prevState.isShown && this.state.isShown) {
```

Comment:
In what case would this block be executed / be needed? When would isShown be flipped from false to true, after `wasClosedByUser === true`?

I think youd only need to handle the controlled, but maybe am missing something you are seeing

Reply:
looked at it again and you are right, doesn't make sense to check for the state flip, updated to just do this for the controlled case